src/value-number.cpp

```diff
@@ -539,6 +539,34 @@ namespace plorth
     }
   }
 
+  template< typename Operation >
+  static void number_op(const ref<context>& ctx, const Operation& op)
+  {
+    ref<number> a;
+    ref<number> b;
+    double result;
+
+    if (!ctx->pop_number(b) || !ctx->pop_number(a))
+    {
+      return;
+    }
+
+    result = op(a->as_real(), b->as_real());
+
+    if (a->is(number::number_type_int) &&
+        b->is(number::number_type_int) &&
+        std::fabs(result) <= INT64_MAX)
+    {
+      // Keep the number as integer.
+      ctx->push_int(static_cast<std::int64_t>(result));
```

Comment:
Casting a `double` into `int64` loses precision. This change makes it pointless to even keep `int64` around.